pkg/utils/gardener/secretsrotation/etcd.go

```diff
@@ -118,13 +118,17 @@ func RewriteEncryptedDataRemoveLabel(
 		message+" (Remove label)",
 		encryptedGVKs...,
 	); err != nil {
-		return err
+		return fmt.Errorf("failed rewriting encrypted data: %w", err)
 	}
 
-	return PatchAPIServerDeploymentMeta(ctx, runtimeClient, namespace, name, func(meta *metav1.PartialObjectMetadata) {
+	err = PatchAPIServerDeploymentMeta(ctx, runtimeClient, namespace, name, func(meta *metav1.PartialObjectMetadata) {
 		delete(meta.Annotations, AnnotationKeyEtcdSnapshotted)
 		delete(meta.Annotations, AnnotationKeyResourcesLabeled)
 	})
+	if err != nil {
+		return fmt.Errorf("failed patching metadata of an API Server deployment: %w", err)
+	}
+	return nil
```

Comment:
Maybe we could do this to follow the usual convention for error checking?
```suggestion
	if err := PatchAPIServerDeploymentMeta(ctx, runtimeClient, namespace, name, func(meta *metav1.PartialObjectMetadata) {
		delete(meta.Annotations, AnnotationKeyEtcdSnapshotted)
		delete(meta.Annotations, AnnotationKeyResourcesLabeled)
	}); err != nil {
		return fmt.Errorf("failed patching metadata of an API Server deployment: %w", err)
	}
```